services/src/main/java/org/keycloak/protocol/oidc/endpoints/UserInfoEndpoint.java

```diff
@@ -124,68 +125,74 @@ public Response issueUserInfoPreflight() {
     @GET
     @NoCache
     public Response issueUserInfoGet(@Context final HttpHeaders headers) {
+        setupCors();
         String accessToken = this.appAuthManager.extractAuthorizationHeaderTokenOrReturnNull(headers);
-        return issueUserInfo(accessToken);
+        authorization(accessToken);
+        return issueUserInfo();
     }
 
     @Path("/")
     @POST
     @NoCache
     public Response issueUserInfoPost() {
+        setupCors();
+        String accessToken;
+
         // Try header first
         HttpHeaders headers = request.getHttpHeaders();
-        String accessToken = this.appAuthManager.extractAuthorizationHeaderTokenOrReturnNull(headers);
+        accessToken = this.appAuthManager.extractAuthorizationHeaderTokenOrReturnNull(headers);
```

Comment:
Declaring `String accessToken` here might be OK instead of declaring it in L139.